File: app/assets/javascripts/ServiceRequest.js
const mongoose = require('mongoose');
const { DataElementSchema } = require('./basetypes/DataElement');
const Code = require('./basetypes/Code');
const Interval = require('./basetypes/Interval');
const Quantity = require('./basetypes/Quantity');
const DateTime = require('./basetypes/DateTime');
const { ComponentSchema } = require('./Component');
const { FacilityLocationSchema } = require('./FacilityLocation');
const { IdSchema } = require('./Id');
const Any = require('./basetypes/Any');

const [Number, String] = [
  mongoose.Schema.Types.Number,
  mongoose.Schema.Types.String,
];

const ServiceRequestSchema = DataElementSchema({
  identifier: Any,
  instantiatesCanonical: Any,
  instantiatesUri: Any,
  basedOn: Any,
  replaces: Any,
  requisition: Any,
  status: Any,
  intent: Any,
  category: Any,
  priority: Any,
  doNotPerform: Any,
  code: Any,
  orderDetail: Any,
  quantity: Any,
  subject: Any,
  encounter: Any,
  occurrence: Any,
  asNeeded: Any,
  authoredOn: Any,
  requester: Any,
  performerType: Any,
  performer: Any,
  locationCode: Any,
  locationReference: Any,
  reasonCode: Any,
  reasonReference: Any,
  insurance: Any,
  supportingInfo: Any,
  specimen: Any,
  bodySite: Any,
  note: Any,
  patientInstruction: Any,
  relevantHistory: Any,
  qdmVersion: { type: String, default: '4.0.0' },
  _type: { type: String, default: 'ServiceRequest' },

});

module.exports.ServiceRequestSchema = ServiceRequestSchema;
module.exports.ServiceRequest = mongoose.model('ServiceRequest', ServiceRequestSchema);

File: app/models/qdm/service_request.rb
module QDM
  # app/models/qdm/service_request.rb
  class ServiceRequest < DataElement
    include Mongoid::Document
    embedded_in :patient
    field :identifier
    field :instantiatesCanonical
    field :instantiatesUri
    field :basedOn
    field :replaces
    field :requisition
    field :status
    field :intent
    field :category
    field :priority
    field :doNotPerform
    field :code
    field :orderDetail
    field :quantity
    field :subject
    field :encounter
    field :occurrence
    field :asNeeded
    field :authoredOn
    field :requester
    field :performerType
    field :performer
    field :locationQDM::Code
    field :locationReference
    field :reasonQDM::Code
    field :reasonReference
    field :insurance
    field :supportingInfo
    field :specimen
    field :bodySite
    field :note
    field :patientInstruction
    field :relevantHistory
    field :qdmVersion, type: String, default: '4.0.0'
  end
end
